### ISSUE TITLE
Collision layers in 4.0 work differently than in 3.5

### ISSUE THREAD

**Issue by lufog** [godotengine/godot#57460]
### Godot version

4.0.alpha.custom_build (78e3e65e7)

### System information

Windows 11 21H2, Vulkan Clustered, Geforce 1660 (511.23)

### Issue description

I'm not sure if it's bug, it just confused me a little.

In Godot 3, `Area2D` (Hurtbox) with collision layers:
![image](https://user-images.githubusercontent.com/10379487/151711682-1fae4a34-4562-43a4-a461-ba3b074ea9af.png)

will detect `Area2D` (Hitbox) with collision layers:
![image](https://user-images.githubusercontent.com/10379487/151711451-66127833-c3ec-4d3f-8619-632a473f720c.png)

But it won't work in Godot 4.

https://user-images.githubusercontent.com/10379487/151713038-6d8f0465-3327-411b-9706-23eecdedd1e6.mp4

### Steps to reproduce

Compare behaviors of two projects.

### Minimal reproduction project

Godot 3 project
[Collision Test G3.zip](https://github.com/godotengine/godot/files/7966671/Collision.Test.G3.zip)

Godot 4 project
[Collision Test G4.zip](https://github.com/godotengine/godot/files/7966672/Collision.Test.G4.zip)

**Comment by Zireael07** [comment]
Likely a consequence of https://github.com/godotengine/godot/pull/50625

Before, if A detected B then B detected A (it was effectively paired). It's no longer the case.

**Comment by lufog** [comment]
@Zireael07 ok, thanks.

**Comment by Zireael07** [comment]
Yep, this is one of the things that WILL catch people unawares, and should be documented somewhere clearly (the other big catch being the change to signals)

**Comment by AttackButton** [comment]
> Yep, this is one of the things that WILL catch people unawares, and should be documented somewhere clearly (the other big catch being the change to signals)

For Areas (2D or 3D) it is working perfectly, better than the old version, the problem is with two ways collisions (CharacterBody2D vs CharacterBody2D). Something that even a good documentation will not prevent the confusion for some situations. This will take time for the users to realize, even after the 4.0 launch.

The issue is about design and usage, not just lack of documentation.
